plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -648,13 +648,19 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             const auto block_deadline = calculate_block_deadline( chain.pending_block_time() );
             push_result pr = push_transaction( block_deadline, trx, persist_until_expired, return_failure_trace, next );
 
-            exhausted = pr.block_exhausted;
+            bool exhausted = pr.block_exhausted;
             if( pr.trx_exhausted ) {
                _unapplied_transactions.add_incoming( trx, persist_until_expired, return_failure_trace, next );
             } else if( pr.persist ) {
                _unapplied_transactions.add_persisted( trx );
             }
 
+            success = !exhausted; // transaction failed if block was exhausted
```

Comment:
For the comment, `failed` is too strong of a word. It was interrupted, it didn't fail. We reserve `fail`/`success` for if the trx asserted or not.

Reply:
done.